algorithms.py

```diff
@@ -947,14 +1331,21 @@ def run_simulate(args):
         population_growth_rates = args.population_growth_rates
     if args.population_sizes is not None:
         population_sizes = args.population_sizes
+    if args.sweep is not None:
+        label_number = 2
+        if num_populations > 1:
+            label_number = 3 ## extra label for other popns
```

Comment:
I don't follow this --- I thought labels and populations were orthogonal?

Reply:
right. this is needed because during a sweep we want to be able to choose recomb events that occur in segments that are population specific. Because of the way that hudson_recombination_event() works, I accomplish this by moving all ancestors in non-sweeping populations (i.e. all but popn 0) into an extra label, and thus I can select segments using the Fenwick tree convention as you had originally done. 

At the beginning of sweep_phase_simulate_until() this happens

https://github.com/kern-lab/msprime/blob/bdd7e2728f2caa0b13821c8a775b195b9d59a1c6/algorithms.py#L629-L645

then this is used for the recombination routines:

https://github.com/kern-lab/msprime/blob/bdd7e2728f2caa0b13821c8a775b195b9d59a1c6/algorithms.py#L816-L821